packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -252,6 +252,38 @@ class CustomPaint extends SingleChildRenderObjectWidget {
 /// By default, [ClipRect] prevents its child from painting outside its
 /// bounds, but the size and location of the clip rect can be customized using a
 /// custom [clipper].
+///
+/// [ClipRect] is commonly used with these widgets, which can paint outside
```

Comment:
This seems a little subtle; we're not implying that all of the other widgets do clip. I can't think of a better way of phrasing the point though.